frame/clearing-house/src/integrations/tests.rs

```diff
@@ -0,0 +1,272 @@
+#![allow(clippy::disallowed_methods)]
+use crate::{Direction::Long, Market, MarketConfig as MarketConfigGeneric};
+use composable_support::validation::Validated;
+use composable_traits::time::ONE_HOUR;
+use frame_support::{assert_ok, pallet_prelude::Hooks};
+use pallet_vamm::VammStateOf;
+use proptest::prelude::*;
+use sp_runtime::{traits::Zero, FixedPointNumber, Percent};
+
+use super::mock::{
+	AccountId, AssetId, Balance, BlockNumber, Decimal, ExtBuilder, MarketId, Moment, Oracle,
+	Origin, Runtime, StalePrice, System, TestPallet, Timestamp, UnsignedDecimal, Vamm, VammId,
+	ALICE, BOB, DOT, PICA, USDC,
+};
+
+impl Default for ExtBuilder {
+	fn default() -> Self {
+		Self {
+			native_balances: vec![],
+			balances: vec![],
+			collateral_type: Some(USDC),
+			max_price_divergence: Decimal::from_inner(i128::MAX),
+		}
+	}
+}
+
+#[test]
+fn externalities_builder_works() {
+	ExtBuilder::default().build().execute_with(|| {});
+}
+
+// -------------------------------------------------------------------------------------------------
+//                                  Helper Functions and Traits
+// -------------------------------------------------------------------------------------------------
+
+fn advance_blocks_by(n: BlockNumber) {
+	for _ in 0..n {
+		let curr_block = System::block_number();
+		if curr_block > 0 {
+			Timestamp::on_finalize(curr_block);
+			System::on_finalize(curr_block);
```

Comment:
I would call `System::on_finalize` last as it is always called last. This probably won't affect functionality now but can down the line

Reply:
same with `on_initialize` but before the other two

Reply:
Ah, thanks, totally missed that